src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -93,8 +93,8 @@ public void addUser(String user) {
     enum Simulation {
         READ_OWN_FILE,
         WRITE_OWN_FILE,
-        READ_SHARED_FILE,
-        WRITE_SHARED_FILE,
+        READ_SHARED_FILE_OR_DIRECTORY,
```

Comment:
Do you think these should be two different things? reading a file is very different to reading a directory. The latter is doing a directory listing.

Reply:
I think they should be different operations

Reply:
ok, I was attempting to make the least amount of changes (no changes to interfaces). The thought being reading/writing actual file is only one step further past checking existence of file/directory.

Reply:
see new code